### DISCUSSION TITLE
any way to listen window resize event?

### DISCUSSION THREAD

**Discussion by ChrisChou-freeman** [beeware/toga#1438] [Q&A] [answered]
hi,  i want Implement adaptive layout, have any way to listen window resize event? 

🙏Thanks

**Comment by freakboy3742** [comment, marked as answer]
No - there isn't currently any window resize event that you can subscribe to.

As far as adaptive layout goes - the long term vision is to replace BeeWare's builtin style language with CSS, and adaptive layouts are a feature supported by CSS.

**Comment by johnzhou721** [comment]
Refs #2304, #2364.

This is currently in the process of being implemented; it's getting *really* close with 1 outstanding major concern left.

**Comment by HalfWhitt** [comment]
This is a pretty late reply, but as of Dec 3, 2025, 0.5.3 (and now 0.5.4) has an [`on_resize`](https://toga.beeware.org/en/stable/reference/api/application/window/#toga.Window.on_resize) handler.
